block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -55,7 +55,22 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
     def latestMessageHash(validator: Validator): F[Option[BlockHash]] =
       latestMessagesMap.get(validator).pure[F]
 
-    def latestMessageHashes: F[Map[Validator, BlockHash]] = latestMessagesMap.pure[F]
+    def latestMessageHashes: F[Map[Validator, BlockHash]] =
+      latestMessagesMap
+        .filter(
+          // This filter is a quick fix until https://github.com/rchain/rchain/issues/3093 is implemented,
+          // as we need to filter latest messages with only active validators. Adn this info is available only
+          // on chain.
+          message => {
+            val messageHeight = heightMap.find(_._2.contains(message._2)) match {
+              // Invalid blocks are not stored in height map
+              case None => invalidBlocksSet.find(_.blockHash == message._2).get.blockNum
+              case m    => m.get._1
+            }
+            messageHeight > (getMaxHeight - 1000)
```

Comment:
I am fine with `Throw away latest messages that are more than X blocks behind` but the value needs to be considered well.

I think the value should be grown if we get more validators. Just do not hard-coded value in the middle of the codes. 
And I am fine with `1000` currently now.


BTW, We also have this `50` hard-coded thing which should be considered well.

Reply:
This is just a fix until we have new PoS with shard config so it should be only temporary. Maybe just creating a variable is enough. We don't expect any change of validators until next epoch.

I agree for deploy threshold of 50 blocks, it is part of shard config.
https://github.com/rchain/rchain/pull/2986/files#diff-0e24b8a374b991ceffedf7d5af4aa696R35

My concern is for slashed validators, if they should be detected and their latest blocks should not be filtered?